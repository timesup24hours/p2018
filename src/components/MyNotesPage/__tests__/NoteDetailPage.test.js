import React from 'react';
import { shallow, mount } from 'enzyme';

import configureStore from 'redux-mock-store';

import thunk from 'redux-thunk';

import ConnectedMyNotesContainer, {
  MyNotesContainer
} from '../MyNotesContainer';

describe('NoteDetailPage', () => {
  const initialState = {
    notes: {
      notes: [],
      currentNote: undefined,
      hasMoreNotesToLoad: true,
      cursor: 0,
      loadAmount: 20,
      loading: false,
      fetchLoading: false,
      message: ''
    }
  };
  const middlewares = [thunk];
  const mockStore = configureStore(middlewares);

  const store = mockStore(initialState);

  const props = {
    notes: [],
    cursor: 0,
    loading: false,
    hasMoreNotesToLoad: true,
    fetchLoading: false,
    message: '',
    loadAmount: 10,
    notesFetchRequested: jest.fn()
  };

  it('should render correctly in "debug" mode', () => {
    const component = shallow(<MyNotesContainer debug />);

    expect(component).toMatchSnapshot();
  });

  it('should render without redux', () => {
    const scrolled = 0;
    const handleScroll = jest.fn();

    const wrapper = mount(
      <MyNotesContainer
        {...props}
        scrolled={scrolled}
        handleScroll={handleScroll}
      />
    );
    expect(wrapper).toMatchSnapshot();

    expect(props.notesFetchRequested).toHaveBeenCalled();
  });

  it('should render with redux', () => {
    const wrapper = mount(<ConnectedMyNotesContainer store={store} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should call the change profile request pending', () => {
    const wrapper = mount(
      <ConnectedMyNotesContainer store={store} {...props} />
    );

    return Promise.resolve(resolve => {
      expect(props.loading).toBeTruthy();
      expect(props.fetchLoading).toBeTruthy();
      resolve(wrapper);
    }).then(() => {
      expect(props.notesFetchRequested).toHaveBeenCalledTimes(1);
      expect(props.notesFetchRequested).toBeCalledWith({
        start: 0,
        loadAmount: 20
      });
      expect(props.loading).toBeFalsy();
      expect(props.fetchLoading).toBeFalsy();
    });
  });
});
