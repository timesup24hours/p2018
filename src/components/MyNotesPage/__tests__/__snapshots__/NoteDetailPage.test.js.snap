// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoteDetailPage should render correctly in "debug" mode 1`] = `
<ForwardRef
  debug={true}
  handleScroll={[Function]}
  scrolled={0}
/>
`;

exports[`NoteDetailPage should render with redux 1`] = `
<Connect(MyNotesContainer)
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <MyNotesContainer
    cursor={0}
    fetchLoading={false}
    hasMoreNotesToLoad={true}
    loadAmount={20}
    loading={false}
    message=""
    notes={Array []}
    notesFetchCancelled={[Function]}
    notesFetchRequested={[Function]}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    storeSubscription={
      Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      }
    }
  >
    <ForwardRef
      cursor={0}
      fetchLoading={false}
      handleScroll={[Function]}
      hasMoreNotesToLoad={true}
      loadAmount={20}
      loading={false}
      message=""
      notes={Array []}
      notesFetchCancelled={[Function]}
      notesFetchRequested={[Function]}
      scrolled={0}
      store={
        Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        }
      }
      storeSubscription={
        Subscription {
          "listeners": Object {
            "clear": [Function],
            "get": [Function],
            "notify": [Function],
            "subscribe": [Function],
          },
          "onStateChange": [Function],
          "parentSub": undefined,
          "store": Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          },
          "unsubscribe": [Function],
        }
      }
    >
      <div
        className="container"
        onScroll={[Function]}
      />
    </ForwardRef>
  </MyNotesContainer>
</Connect(MyNotesContainer)>
`;

exports[`NoteDetailPage should render without redux 1`] = `
<MyNotesContainer
  cursor={0}
  fetchLoading={false}
  handleScroll={[MockFunction]}
  hasMoreNotesToLoad={true}
  loadAmount={10}
  loading={false}
  message=""
  notes={Array []}
  notesFetchRequested={
    [MockFunction] {
      "calls": Array [
        Array [
          Object {
            "loadAmount": 20,
            "start": 0,
          },
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  scrolled={0}
>
  <ForwardRef
    cursor={0}
    fetchLoading={false}
    handleScroll={[Function]}
    hasMoreNotesToLoad={true}
    loadAmount={10}
    loading={false}
    message=""
    notes={Array []}
    notesFetchRequested={
      [MockFunction] {
        "calls": Array [
          Array [
            Object {
              "loadAmount": 20,
              "start": 0,
            },
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      }
    }
    scrolled={0}
  >
    <div
      className="container"
      onScroll={[Function]}
    />
  </ForwardRef>
</MyNotesContainer>
`;
